functions/controllers/onboardingController.js

```diff
@@ -304,15 +334,20 @@ const setupUserProfile = functions.https.onCall(async (data, context) => {
       occupation: occupation,
       socialLink: socialLinks,
       bio: bio,
+      profilePhotoUrl: profilePhotoUrl,
     };
 
     logger.log('Updating user document in Firestore', userDoc);
 
     await userRef.set(userDoc, { merge: true });
-
+    // Call the Onboarding Status Transition Utility function to advance the status
+    const onboardingResult = await progressOnboardingStatus(uid);
```

Comment:
Added this call to advance onboarding status, which I did not do previously sorry!